vllm/model_executor/layers/fused_moe/layer.py

```diff
@@ -132,17 +138,18 @@ def forward_tpu(
             num_expert_group: Optional[int] = None,
             custom_routing_function: Optional[Callable] = None
     ) -> torch.Tensor:
-        from vllm.model_executor.layers.fused_moe.moe_pallas import fused_moe
         assert not use_grouped_topk
         assert num_expert_group is None
         assert topk_group is None
         assert custom_routing_function is None
-        return fused_moe(hidden_states=x,
-                         w1=layer.w13_weight,
-                         w2=layer.w2_weight,
-                         topk=top_k,
-                         gating_output=router_logits,
-                         renormalize=renormalize)
+        return fused_moe_pallas(hidden_states=x,
+                                w1=layer.w13_weight,
+                                w2=layer.w2_weight,
+                                topk=top_k,
+                                gating_output=router_logits,
+                                renormalize=renormalize)
+
+    forward_native = forward_cuda
```

Comment:
What is this for?

Reply:
when we use inductor, we will use `forward_native` . this function needs to be implemented.

Reply:
you can also write a pytorch native implementation, but I doubt if inductor can optimize it. that's why I use `forward_cuda` as `forward_native`

Reply:
basically, `forward_native` is the function we compile when we use inductor.

Reply:
I see. I was just surprised because I didn't know that a class method can be defined this way

Reply:
So it's essentially the same as

```python
def forward_native(self, *args, **kwargs):
    return self.forward_cuda(*args, **kwargs)
```
?

Reply:
yes.